Apps/System/settings.c

```diff
@@ -0,0 +1,196 @@
+    bluetooth_bond_acknowledge(1);
+    window_stack_pop(false);
+}
+
+static void _bt_pair_do_reject(ClickRecognizerRef recognizer, void *context) {
+    APP_LOG("settings", APP_LOG_LEVEL_INFO, "BT pair reject!");
+    bluetooth_bond_acknowledge(0);
+    window_stack_pop(false);
+}
+
+static void _bt_pair_action_click_config_provider(void *context) {
+    window_single_click_subscribe(BUTTON_ID_UP, (ClickHandler) _bt_pair_do_accept);
+    window_single_click_subscribe(BUTTON_ID_DOWN, (ClickHandler) _bt_pair_do_reject);
+}
+
+static void _bt_pair_update_proc(Window *window, GContext *ctx) {
+    graphics_context_set_text_color(ctx, GColorBlack);
+    graphics_draw_text(ctx, "Pair?", fonts_get_system_font(FONT_KEY_GOTHIC_28_BOLD), GRect(0, 40, DISPLAY_COLS-ACTION_BAR_WIDTH, 28), GTextOverflowModeTrailingEllipsis, GTextAlignmentCenter, 0);
+    graphics_draw_text(ctx, _bt_pair_name, fonts_get_system_font(FONT_KEY_GOTHIC_14), GRect(0, 100, DISPLAY_COLS-ACTION_BAR_WIDTH, 24), GTextOverflowModeTrailingEllipsis, GTextAlignmentCenter, 0);
+}
+
+static void _bt_pair_window_load(Window *window) {
+    _bt_pair_accept = gbitmap_create_with_resource(RESOURCE_ID_CHECK_BLACK);
+    _bt_pair_reject = gbitmap_create_with_resource(RESOURCE_ID_DISMISS_BLACK);
+    GRect bmsize = gbitmap_get_bounds(_bt_pair_accept);
+    APP_LOG("settings", APP_LOG_LEVEL_INFO, "bt pair window load");
+    
+    window_set_background_color(window, GColorWhite);
+    
+    Layer *window_layer = window_get_root_layer(window);
+    GRect bounds = layer_get_unobstructed_bounds(window_layer);
+    
+    _bt_pair_layer = layer_create(bounds);
+    layer_add_child(window_layer, _bt_pair_layer);
+    layer_set_update_proc(_bt_pair_layer, _bt_pair_update_proc);
+    layer_mark_dirty(_bt_pair_layer);
+    
+    _bt_pair_action_bar = action_bar_layer_create();
+    action_bar_layer_set_click_config_provider(_bt_pair_action_bar, _bt_pair_action_click_config_provider);
+    action_bar_layer_set_icon_animated(_bt_pair_action_bar, BUTTON_ID_UP, _bt_pair_accept, true);
+    action_bar_layer_set_icon_animated(_bt_pair_action_bar, BUTTON_ID_DOWN, _bt_pair_reject, true);
+    action_bar_layer_set_background_color(_bt_pair_action_bar, GColorBlack);
+    action_bar_layer_add_to_window(_bt_pair_action_bar, window);
+}
+
+static void _bt_pair_window_unload(Window *window) {
+    action_bar_layer_destroy(_bt_pair_action_bar);
+    gbitmap_destroy(_bt_pair_accept);
+    gbitmap_destroy(_bt_pair_reject);
+    layer_destroy(_bt_pair_layer);
+    status_bar_layer_destroy(_bt_pair_status);
+    free(_bt_pair_name);
+}
+
+static Window *_main_window;
+static Menu *_menu;
+static StatusBarLayer *status_bar;
+
+static void _reset_menu_items(void);
+
+
+static void exit_to_watchface(struct Menu *menu, void *context)
+{
+    /* Exit to watchface */
+    hw_bluetooth_advertising_visible(0);
+    window_stack_pop(true);
+}
+
+static struct MenuItems *_really_wipe_fs(const struct MenuItem *ctx) {
```

Comment:
#ifdef for platform support?

Reply:
This should be safe on Snowy too?  If it isn't, we oughta fix it.